src/pybind/mgr/prometheus/module.py

```diff
@@ -1283,6 +1283,8 @@ def get_rbd_stats(self) -> None:
         # '*' can be used to indicate all pools or namespaces
         pools_string = cast(str, self.get_localized_module_option('rbd_stats_pools'))
         pool_keys = []
+        osd_map = self.get('osd_map')
```

Comment:
Great (and easy) optimization! `osd_map` was a great performance killer in large cluster, so this is a quick win!

Reply:
That said... if the only purpose of fetching the osd_map is obtaining the list of pools... Why not using `get('df')`? `len(osdmap)` ~ number of OSDs (stretch goal is 4-8 kOSDs), while `len(df)` ~ number of pools (which is in the range of dozens).